### ISSUE TITLE
redundant_pattern_matching shouldn't complain about macro generated redundant patterns

### ISSUE THREAD

**Issue by xfix** [rust-lang/rust-clippy#6065]
I tried this code:

```rust
fn main() {
    macro_rules! if_let_in_macro {
        ($pat:pat, $x:expr) => {
            if let Some($pat) = $x {}
        };
    }

    if_let_in_macro!(_, Some(42));
}
```

Macro is external to the crate in actual code.

I expected to see this happen: No complaints

Instead, this happened: 

```
warning: redundant pattern matching, consider using `is_some()`
 --> src/main.rs:4:20
  |
4 |               if let Some($pat) = $x {}
  |  _____________-      ^^^^^^^^^^
5 | |         };
6 | |     }
7 | |
8 | |     if_let_in_macro!(_, Some(42));
  | |     ------------------------------
  | |_____|__________________________|
  |       |                          help: try this: `if Some(42).is_some()`
  |       in this macro invocation
  |
  = note: `#[warn(clippy::redundant_pattern_matching)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching
  = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

I believe this is wrong as it's not always fixable in macros

### Meta

- `cargo clippy -V`: clippy 0.0.212 (f3c923a 2020-09-17)
- `rustc -Vv`:
  ```
  rustc 1.48.0-nightly (f3c923a13 2020-09-17)
  binary: rustc
  commit-hash: f3c923a13a458c35ee26b3513533fce8a15c9c05
  commit-date: 2020-09-17
  host: x86_64-unknown-linux-gnu
  release: 1.48.0-nightly
  LLVM version: 11.0
  ```

<details><summary>Backtrace</summary>
  <p>
  
  ```
  <backtrace>
  ```
  
  </p>
</details>
